Improve ci

rffontenelle opened the pull request:
- **Use the new Transifex CLI  toll:** The Python implementation named transifex-client uses Transifex's APIv2, which is deprecated and will sunset in november 2022. The new tool, called Transifex CLI, is written in Golang and is APIv3-ready. So adjusting the CI and requirements.txt is required.
- **Fix condition in the merge job]:** Having the proper condition is important to have the translation
update in the latest Python version (python-newest project), and
the condition wouldn't match the current repository.

rffontenelle committed:
Use new Transifex CLI tool

The Python implementation named transifex-client uses Transifex's
APIv2, which is deprecated and will sunset in november 2022.

The new tool, called Transifex CLI, is written in Golang and is
APIv3-ready. So adjusting the CI and requirements.txt is required.

rffontenelle committed:
Fix condition in the merge job

Having the proper condition is important to have the translation
update in the latest Python version (python-newest project), and
the condition wouldn't match the current repository.

m-aciek commented on .github/workflows/update.yml:
```diff
@@ -35,6 +35,10 @@ jobs:
+      - name: Install Transifex CLI
+        run: |
+          curl -o- https://raw.githubusercontent.com/transifex/cli/master/install.sh | bash
+          mv tx /usr/local/bin/tx
```
As far as I'm concerned this will leave a LICENSE file coming from new tx client in the working directory.

Consider installing in `/user/local/bin` directly.
```suggestion
        run: |
          curl -o- https://raw.githubusercontent.com/transifex/cli/master/install.sh | bash
        working-directory: /usr/local/bin
```

m-aciek reviewed (changes requested):
~~New Transifex CLI doesn't support `TX_TOKEN` environment variable which `update.sh`/GH Action uses. [1]~~

~~We should update this line [2] to contain the Transifex API token.~~

~~```~~
~~tx --token ${TX_TOKEN} pull -l ${LANGUAGE} --use-git-timestamps --parallel~~
~~```~~

~~[1] https://github.com/transifex/cli/issues/116~~
~~[2] https://github.com/python/python-docs-ja/blob/3.10/scripts/update.sh#L42~~

EDIT: One comment. Otherwise looks good to me 👍 .

rffontenelle committed:
Set a working-directory to install tx tool

To avoid having LICENSE in checkout directory

Co-authored-by: Maciej Olko <maciej.olko@affirm.com>

rffontenelle commented:
@m-aciek Thanks for the suggestion.

m-aciek commented:
Possibly we also need to change format of .tx/config file for new CLI. Is it generated automatically?

rffontenelle commented:
@m-aciek Currently .tx/config is recreated when scripts/update.sh runs the sed command outputting to .tx/config. Still, some changes are necessary to fit the new tx client, so thanks for bringing this up.

rffontenelle committed:
Update sphinx-intl required version

Last released sphinx-intl uses old transifex-client, which uses
the deprecated Transifex APIv2. There is a fix committed, but
it is unreleased. So using specific commit where new Transifex
CLI tool works.

rffontenelle committed:
Update script to work with the new TX command

Previous solution of patching .tx/config does not work with the
new client, so the script had to be updated with new solution.
Now, translations are pulled into a sub-directory of cpython, and
then installed in the languages's repository.

Also using 'realpath' tool in ROOTDIR to retrieve absolute path,
which actually eases referencing the repository root directory.

rffontenelle committed:
Update actions versions

One of the reason is that some of them use node.js 12, which is
now deprecated. Actions is displaying warnings about it.

v4 of setup-python is particular useful because it introduces
package caching, which reduces the time of workflow run.

rffontenelle committed:
Disable pushing translations

Source files auto-update is now done elsewhere

rffontenelle commented:
The last commit disables pushing source files to Transifex (`tx push`) because translations are now updated via [rffontenelle/docspush-transifex](https://github.com/rffontenelle/docspush-transifex), as discussed in [Python's translation mailing list](https://mail.python.org/archives/list/translation@python.org/thread/D2UA35BS2SYX522VKUVWLB7HXPHOYRWQ/).
